source/game/field/ObjectDirector.cc

```diff
@@ -0,0 +1,55 @@
+#include "ObjectDirector.hh"
+
+#include "game/system/CourseMap.hh"
+#include "game/system/map/MapdataGeoObj.hh"
+
+namespace Field {
+
+ObjectDirector *ObjectDirector::CreateInstance() {
+    assert(!s_instance);
+    s_instance = new ObjectDirector;
+
+    s_instance->createObjects();
+
+    return s_instance;
+}
+
+void ObjectDirector::DestroyInstance() {
+    assert(s_instance);
+    delete s_instance;
+    s_instance = nullptr;
+}
+
+ObjectDirector *ObjectDirector::Instance() {
+    return s_instance;
+}
+
+ObjectDirector::ObjectDirector() = default;
+
+ObjectDirector::~ObjectDirector() = default;
+
+void ObjectDirector::createObjects() {
+    const auto *courseMap = System::CourseMap::Instance();
+    size_t objectCount = courseMap->getGeoObjCount();
+    m_objects.reserve(objectCount);
+
+    for (size_t i = 0; i < objectCount; ++i) {
+        const auto *pObj = courseMap->getGeoObj(i);
+        assert(pObj);
```

Comment:
If you're just going to assert on every call to `getGeoObj` maybe it makes sense to just throw the assert in `getGeoObj` itself? I understand if you oppose that though, because then it may appear as if there is no `nullptr` sanitation here.

Reply:
We do not want to do this in the event that we actually *do* check for `nullptr` in these returns.

Reply:
Oh good point